profile-widget/profilescene.cpp

```diff
@@ -627,3 +625,19 @@ void ProfileScene::draw(QPainter *painter, const QRect &pos,
 	}
 	painter->drawImage(pos, image);
 }
+
+// Calculate the new zoom position when the mouse is dragged by delta.
+// This is annoyingly complex, because the zoom position is give as
+// a real between 0 and 1.
+double ProfileScene::calcZoomPosition(double zoom, double originalPos, double delta)
+{
+	double factor = 1.0 - 1.0/zoom;
+	if (fabs(factor) < 1e-10)
+		return 0.0;
```

Comment:
please use `IS_FP_SAME(factor, 0.0)`
For reasons I cannot comprehend that is now in `subsurface_string.h`

Reply:
> please use `IS_FP_SAME(factor, 0.0)`

This macro invocation is silly. It expands to `fabs((factor) <= 0.000001 * fabs(factor)`, which is never `true` for `factor` unequal 0.0 unless you reach resolution limits (or factor is effectively 0). This seems like a very bad idea to me given the different precision of standard floating point registers and SIMD registers.

Reply:
Actually - I don't see how this can ever be true for `factor` which is not precisely 0.0, which makes this whole thing pretty pointless.

Reply:
We've had this code since 2014 - introduced in 2f822ec0dcd2d715bf4234023b510aed9fe90f0b by @atdotde 

Since this is the type of thing that I prefer to leave to the experts - can the two of you please figure out what a reasonable way is to compare floating point values?

Reply:
The problem was introduced in b6a30dcdd3225e67e7eed7ead68c919c36243ffc. This fails when one of the arguments is 0. Note that there are other occurrences that have to be fixed.

Reply:
well - I don't want to change the macro without hearing Robert, so let's just leave your code as is and move on.
I still need to find time to test it...